[chore] activate shellcheck in `lint` target

sultanowskii opened the pull request:
Resolves https://github.com/open-telemetry/opentelemetry-injector/issues/74

basti1302 reviewed (approved):
Awesome! Thanks so much for contributing this.

basti1302 commented on packaging/fpm/deb/build.sh:
```diff
@@ -5,8 +5,10 @@
-. $SCRIPT_DIR/../common.sh
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# shellcheck disable=SC1091 # Including common.sh
```
Actually, nevermind, since common.sh also is found by the pattern, it is inspected by shellcheck anyway, so this is fine as it is.

sultanowskii commented on packaging/fpm/common.sh:
```diff
@@ -70,8 +71,8 @@ download_nodejs_agent() {
-    npm pack @opentelemetry/auto-instrumentations-node@${tag#v}
-    mv *.tgz otel-js.tgz
+    npm pack "@opentelemetry/auto-instrumentations-node@${tag#v}"
+    mv ./*.tgz otel-js.tgz
```
Yes, https://www.shellcheck.net/wiki/SC2035

sultanowskii commented:
Regarding the issues CI has ran into - I couldn't reproduce them on my machine - `make lint` just doesn't give me these warnings

I've looked into it - looks like in CI, shellcheck is of some old version, while I have 0.11.0

Example (https://www.shellcheck.net/wiki/SC2002), quote:
> In later versions, it is [optional](https://www.shellcheck.net/wiki/optional). It must be explicitly enabled with a [directive](https://www.shellcheck.net/wiki/directive) enable=useless-use-of-cat

So for now I've added `enable-useless-use-of-cat` for local vs CI compliance and added the explicit ignoring of `SC2317`, but it might be a good idea to bump its version (though, I have no clue where is it being installed - it isn't mentioned in .github workflows at all)
